### ISSUE TITLE
While deleting product there is a bug

### ISSUE THREAD

**Issue by SDiamante13** [SDiamante13/ordering-system#20]
TLDR; Deletes should be cascaded operations 

Hibernate: delete from products where product_id=?
 WARN 4 --- [io-21821-exec-2] o.h.engine.jdbc.spi.SqlExceptionHelper: SQL Error: 0, SQLState: 23503
 ERROR 4 --- [io-21821-exec-2] o.h.engine.jdbc.spi.SqlExceptionHelper: ERROR: update or delete on table "products" violates foreign key constraint "fko6helt0ucmegaeachjpx40xhe" on table "order_product"
 Detail: Key (product_id)=(6) is still referenced from table "order_product".
 INFO 4 --- [io-21821-exec-2] o.h.e.j.b.internal.AbstractBatchImpl: HHH000010: On release of batch it still contained JDBC statements
 ERROR 4 --- [io-21821-exec-2] o.h.i.ExceptionMapperStandardImpl: HHH000346: Error during managed flush [org.hibernate.exception.ConstraintViolationException: could not execute statement]

**Comment by SDiamante13** [comment]
Or you shouldn't be able to delete a product. You can just decrease the quantity in stock to 0
